2023-09-18 17:45:41.319460: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:45:41.319628: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:45:48,930 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:45:49,004 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:45:49,004 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:45:49,004 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:45:49,111 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:45:49,111 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,127 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.41}
INFO flwr 2023-09-18 17:45:49,127 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:45:49,388 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:45:49,410 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:45:49,412 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,420 | server.py:125 | fit progress: (1, 0.41949927693996175, {'accuracy': 0.7933333333333333}, 0.2931229000000002)
DEBUG flwr 2023-09-18 17:45:49,421 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:45:49,443 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:45:49,443 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:45:49,443 | server.py:223 | fit_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:49,475 | server.py:237 | fit_round 2 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,487 | server.py:125 | fit progress: (2, 0.42349204893953535, {'accuracy': 0.79}, 0.3590221000000007)
DEBUG flwr 2023-09-18 17:45:49,487 | server.py:173 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:49,510 | server.py:187 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,511 | server.py:223 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:45:49,540 | server.py:237 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,551 | server.py:125 | fit progress: (3, 0.42346371223441065, {'accuracy': 0.7933333333333333}, 0.42414090000000115)
DEBUG flwr 2023-09-18 17:45:49,551 | server.py:173 | evaluate_round 3: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:45:49,573 | server.py:187 | evaluate_round 3 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,573 | server.py:223 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:45:49,603 | server.py:237 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,615 | server.py:125 | fit progress: (4, 0.42264233037030796, {'accuracy': 0.79}, 0.48764820000000064)
DEBUG flwr 2023-09-18 17:45:49,615 | server.py:173 | evaluate_round 4: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,640 | server.py:187 | evaluate_round 4 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,641 | server.py:223 | fit_round 5: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,679 | server.py:237 | fit_round 5 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,690 | server.py:125 | fit progress: (5, 0.422373243983484, {'accuracy': 0.79}, 0.562880100000001)
DEBUG flwr 2023-09-18 17:45:49,691 | server.py:173 | evaluate_round 5: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,724 | server.py:187 | evaluate_round 5 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,724 | server.py:223 | fit_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,763 | server.py:237 | fit_round 6 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,775 | server.py:125 | fit progress: (6, 0.4223754363533426, {'accuracy': 0.79}, 0.6476857000000003)
DEBUG flwr 2023-09-18 17:45:49,775 | server.py:173 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,804 | server.py:187 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,805 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,844 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,858 | server.py:125 | fit progress: (7, 0.4223754286299732, {'accuracy': 0.79}, 0.7311234000000013)
DEBUG flwr 2023-09-18 17:45:49,859 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,883 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,884 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,918 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:49,930 | server.py:125 | fit progress: (8, 0.4223754285737013, {'accuracy': 0.79}, 0.802676700000001)
DEBUG flwr 2023-09-18 17:45:49,930 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,957 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:49,957 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:49,993 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:50,004 | server.py:125 | fit progress: (9, 0.4223754285733582, {'accuracy': 0.79}, 0.8772373000000009)
DEBUG flwr 2023-09-18 17:45:50,004 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:50,037 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:45:50,037 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:50,083 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:45:50,095 | server.py:125 | fit progress: (10, 0.42237542857335814, {'accuracy': 0.79}, 0.968069400000001)
DEBUG flwr 2023-09-18 17:45:50,096 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:45:50,127 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:45:50,128 | server.py:147 | FL finished in 1.0002292000000015
INFO flwr 2023-09-18 17:45:50,128 | app.py:218 | app_fit: losses_distributed [(1, 0.4173398111015558), (2, 0.4164774010578791), (3, 0.4203035526332401), (4, 0.42046565456049784), (5, 0.4202444425651005), (6, 0.420244425535202), (7, 0.42024442851543425), (8, 0.42024442851543425), (9, 0.42024442851543425), (10, 0.42024442851543425)]
INFO flwr 2023-09-18 17:45:50,128 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:45:50,128 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:45:50,128 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.41949927693996175), (2, 0.42349204893953535), (3, 0.42346371223441065), (4, 0.42264233037030796), (5, 0.422373243983484), (6, 0.4223754363533426), (7, 0.4223754286299732), (8, 0.4223754285737013), (9, 0.4223754285733582), (10, 0.42237542857335814)]
INFO flwr 2023-09-18 17:45:50,128 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.41), (1, 0.7933333333333333), (2, 0.79), (3, 0.7933333333333333), (4, 0.79), (5, 0.79), (6, 0.79), (7, 0.79), (8, 0.79), (9, 0.79), (10, 0.79)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.83601448,  1.03442756,  0.76667564,  0.29334895,  0.2003213 ,
         0.05090274, -0.01609389,  0.17164324]]), array([-0.13799345])]
[array([[-1.83601448,  1.03442756,  0.76667564,  0.29334895,  0.2003213 ,
         0.05090274, -0.01609389,  0.17164324]]), array([-0.13799345])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90717918,  0.99117265,  0.82823773,  0.31180758,  0.20529287,
         0.04099708, -0.05117802,  0.08132428]]), array([-0.05425296])]
[array([[-1.90717918,  0.99117265,  0.82823773,  0.31180758,  0.20529287,
         0.04099708, -0.05117802,  0.08132428]]), array([-0.05425296])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90463898,  0.98992094,  0.82727792,  0.31188772,  0.20477552,
         0.041457  , -0.05157728,  0.08049841]]), array([-0.05299261])]
[array([[-1.90463898,  0.98992094,  0.82727792,  0.31188772,  0.20477552,
         0.041457  , -0.05157728,  0.08049841]]), array([-0.05299261])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.87993017,  0.9799194 ,  0.80980526,  0.34833738,  0.20346637,
         0.04040037, -0.05741211,  0.06473286]]), array([-0.06022272])]
[array([[-1.87993017,  0.9799194 ,  0.80980526,  0.34833738,  0.20346637,
         0.04040037, -0.05741211,  0.06473286]]), array([-0.06022272])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007968,  0.96024013,  0.81167996,  0.3860303 ,  0.18078683,
         0.02386472, -0.03219603,  0.05729239]]), array([-0.03548861])]
[array([[-1.89007968,  0.96024013,  0.81167996,  0.3860303 ,  0.18078683,
         0.02386472, -0.03219603,  0.05729239]]), array([-0.03548861])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007956,  0.96026474,  0.8116639 ,  0.3859392 ,  0.18082944,
         0.02387834, -0.03213027,  0.05726342]]), array([-0.03552772])]
[array([[-1.89007956,  0.96026474,  0.8116639 ,  0.3859392 ,  0.18082944,
         0.02387834, -0.03213027,  0.05726342]]), array([-0.03552772])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552742])]
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552742])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552743])]
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552743])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552743])]
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552743])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552743])]
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552743])]
>>>>>>>>>>>>>run time:
0.40625s
